### PR TITLE
Added unknown variable parent and loop management

### PR DESCRIPTION
**Pull request by alexisszmundy**
Variables not in specs are now linked to their parent if they have a "_" delimiter in their name

### CODE REVIEW

File: src/main/java/fr/insee/genesis/controller/utils/LoopIdentifier.java [hunk 1]
```diff
@@ -15,6 +18,18 @@ private LoopIdentifier() {
 	public static String getLoopIdentifier(String variableName, VariablesMap variablesMap, int index) {
 		Variable variable = variablesMap.getVariable(variableName);
 		if (variable == null) {
+			if(variableName.startsWith(Constants.FILTER_RESULT_PREFIX)
+					&& variablesMap.hasVariable(variableName.replace(Constants.FILTER_RESULT_PREFIX,""))
+			){
+				Variable parentVariable = variablesMap.getVariable(variableName.replace(Constants.FILTER_RESULT_PREFIX,""));
```
Comment (alicela) [inline src/main/java/fr/insee/genesis/controller/utils/LoopIdentifier.java:24]: I don't understand the wording "parentvariable". It's not the parent as it's not a tree. It's more a "linkedvariable" or "attachedVariable"...

File: src/main/java/fr/insee/genesis/controller/utils/LoopIdentifier.java [hunk 2]
```diff
@@ -15,6 +18,18 @@ private LoopIdentifier() {
...
+				return parentVariable.getGroupName();
+			}
+			if(variableName.endsWith(Constants.MISSING_SUFFIX)
```
Comment (alicela) [inline src/main/java/fr/insee/genesis/controller/utils/LoopIdentifier.java:27]: this "if" is exactly the same as above. Maybe create a method "getGroupNameForSpecificVariable(String constant to replace, beginOrEnd, variableName) ?

File: src/main/java/fr/insee/genesis/controller/utils/LoopIdentifier.java [hunk 3]
```diff
@@ -26,7 +41,25 @@ public static String getLoopIdentifier(String variableName, VariablesMap variabl
...
+		List<String> varsEno = Arrays.asList(Constants.getEnoVariables());
+		if ( variable == null ) {
+			if(varsEno.contains(variableName))
+			{
+				return Constants.ROOT_GROUP_NAME;
+			}
+			if(variableName.startsWith(Constants.FILTER_RESULT_PREFIX)
+					&& variablesMap.hasVariable(variableName.replace(Constants.FILTER_RESULT_PREFIX,""))
+			){
+				return variableName.replace(Constants.FILTER_RESULT_PREFIX,"");
```
Comment (alicela) [inline src/main/java/fr/insee/genesis/controller/utils/LoopIdentifier.java:53]: you return the variable and not the group

Comment (alexisszmundy) [reply src/main/java/fr/insee/genesis/controller/utils/LoopIdentifier.java:53]: The name of the function is misleading, renamed it to "getParentName"

File: src/main/java/fr/insee/genesis/controller/utils/LoopIdentifier.java (hunk 1 above)
Comment (alicela) [reply src/main/java/fr/insee/genesis/controller/utils/LoopIdentifier.java]: Finally, we choose "relatedVariable", are you ok ?

Comment (alexisszmundy) [reply src/main/java/fr/insee/genesis/controller/utils/LoopIdentifier.java]: yes

**Pull request merged**